src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2641,16 +2771,198 @@ public override object PayloadValue(int index)
+        private int m_count;
+        private long m_size;
+        #endregion
+    }
+
+    /// <summary>
+    /// BucketKind is used to describe the type of the GCFitBucketInfo events.
+    /// </summary>
+    public enum BucketKind
+    {
+        /// <summary>
+        /// When the BucketKind is LargestFreeListItems, the free list items in Gen2 are counted and reported
+        /// The counting has a couple of heuristics that avoid spending too much time counting.
+        /// </summary>
+        LargestFreeListItems = 0,
+
+        /// <summary>
+        /// When the BucketKind is PlugsInCondemned, the buckets in the associated GCFitBucketInfo events
+        /// represents plugs that we found during the plan phase of a Gen1 GC where it is planned to fit in the 
+        /// current (i.e. condemned) generation. 
+        /// </summary>
+        PlugsInCondemned = 1
+    }
+
+    public class GCFitBucketInfoTraceData : TraceEvent
+    {
+        public int ClrInstanceID { get { return GetInt16At(0); } }
+
+        /// <summary>
+        /// The kind of the buckets (see the comments for the enum BucketKind for more information)
+        /// </summary>
+        public BucketKind BucketKind { get { return (BucketKind)GetInt16At(2); } }
+
+        /// <summary>
+        /// The total size of all buckets.
+        /// </summary>
+        public long TotalSize { get { return GetInt64At(4); } }
+
+        /// <summary>
+        /// The number of buckets.
+        /// </summary>
+        public int Count { get { return GetInt16At(12); } }
+
+        public GCFitBucket Buckets(int bucketIndex)
+        {
+            Debug.Assert(0 <= bucketIndex && bucketIndex < Count);
+            int baseAddress = 14 + SizeOfGCFitBucket * bucketIndex;
+            int index = GetInt32At(baseAddress);
+            int count = GetInt32At(baseAddress + 4);
+            long size = (PointerSize == 8) ? GetInt64At(baseAddress + 8) : (long)GetInt32At(baseAddress + 8);
+            return new GCFitBucket(index, count, size);
+        }
+
+
+        #region Private
+        internal GCFitBucketInfoTraceData(Action<GCFitBucketInfoTraceData> action, int eventID, int task, string taskName, Guid taskGuid, int opcode, string opcodeName, Guid providerGuid, string providerName)
+            : base(eventID, task, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName)
+        {
+            Action = action;
+        }
+        protected internal override void Dispatch()
+        {
+            Action(this);
+        }
+        protected internal override Delegate Target
+        {
+            get { return Action; }
+            set { Action = (Action<GCFitBucketInfoTraceData>)value; }
+        }
+        protected internal override void Validate()
+        {
+            int size0 = 14 + Count * SizeOfGCFitBucket;
+            Debug.Assert(!(Version == 0 && EventDataLength != size0));
+            Debug.Assert(!(Version > 0 && EventDataLength < size0));
```

Comment:
there's only version 0 so I don't understand what this assert is supposed to achieve?

Reply:
It seems to be the way how other `Validate` method works. My interpretation is that whenever we have a newer version, the newer version should have the earlier layout keep unchanged, that's why from a size perspective it should be at least as large as the earlier version.